### DISCUSSION TITLE
More control over logs verbosity ?

### DISCUSSION THREAD

**Discussion by rplauta** [flant/shell-operator#391] [Q&A]
Hello, I was wondering if there is a way to manage the amount of logs other than LOG_LEVEL and LOG_NO_TIME Env-Variable ?

I'd like to have a middle ground between only seeing errors and having a bare minimum of infos that I echo / print myself  from the hook script, unfortunately the latest will only show up if log_level is set to info which will add extra log lines for me to filter later down the road.

Example below, hook triggers will logs 4 lines and only the 2nd is of interest to me.

```json
{"binding":"kubernetes","event":"kubernetes","hook":"events_hook.sh","level":"info","msg":"Execute hook","queue":"main","task":"HookRun","time":"2022-05-04T12:14:48Z"}

{"binding":"kubernetes","event":"kubernetes","hook":"events_hook.sh","level":"info","msg":"[ACTION] Terminating failed-pod which completed with exitcode 2 : Error","output":"stdout","queue":"main","task":"HookRun","time":"2022-05-04T12:14:49Z"}

{"binding":"kubernetes","event":"kubernetes","hook":"events_hook.sh","level":"info","msg":"pod \"failed-pod\" deleted","output":"stdout","queue":"main","task":"HookRun","time":"2022-05-04T12:14:49Z"}

{"binding":"kubernetes","event":"kubernetes","hook":"events_hook.sh","level":"info","msg":"Hook executed successfully","queue":"main","task":"HookRun","time":"2022-05-04T12:14:49Z"}

{"binding":"kubernetes","event.id":"6050fb6d-df48-49ba-ba1b-6fd6f670419a","level":"info","msg":"queue task HookRun:main:kubernetes:events_hook.sh:kubernetes","operator.component":"handleEvents","queue":"main","task.id":"c2103ffc-a061-4d53-ba0e-3e563100f7d9","time":"2022-05-04T12:14:50Z"}
```

Any way to have more controls over my hook logs ? Thanks

**Comment by diafour** [comment]
Hello! If your hooks print JSON lines, you can use "proxying JSON logs" available since version 1.0.10. In short, it is possible to set LOG_LEVEL=error and see only errors from the shell-operator and also get full control over log level for your hooks.  Check new section in [RUNNING.md](https://github.com/flant/shell-operator/blob/main/RUNNING.md#notes-on-json-log-proxying).

**Reply by rplauta** [reply]
Thanks for your reply, I did check the --log-proxy-hook-json flag as well as the commit that added it but I can't seem to get a notable difference in my output when setting LOG_PROXY_HOOK_JSON to true apart from it giving me a similar output to what I would have by setting LOG_LEVEL to info.

May you be able to provide a code snipet of some hook code that would make proper use of this feature ?

Thank you for your time.
